File: 品味Java的21種設計模式/dp_cpp/command/example8/PrintService.cpp
#include "PrintService.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace command
			{
				namespace example8
				{

					PrintService::PrintService(std::string s)
					{
						InitializeInstanceFields();
						str = s;
					}

					void PrintService::execute()
					{
						puts("打印的内容为="+str);
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/command/example8/PrintService.java
package cn.javass.dp.command.example8;

public class PrintService implements Command {
    /**
     * 要输出的内容
     */
    private String str = "";
    /**
     * 构造方法，传入要输出的内容
     * @param s 要输出的内容
     */
    public PrintService(String s) {
        str = s;
    }

    public void execute() {
        System.out.println("打印的内容为=" + str);
    }
}